src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthDirectives.scala

```diff
@@ -23,24 +24,50 @@ trait OAuthDirectives {
   def authenticationToken(requestDescription: String): Directive1[String] =
     headerValue(authorization()) | reject(CredentialsMissingRejection(requestDescription))
 
-  def authenticate(token: String, requestDescription: String, authService: AuthService): Directive1[AuthenticatedUser] =
-    authService.authenticate(token) match {
-      case Success(authUser) => provide(authUser)
-      case Failure(ex) => {
-        logger.error(s"Authentication failed with exception: ${ex.getMessage}")
-        reject(CredentialsRejectedRejection(requestDescription))
+  def authenticate(token: String, requestDescription: String, authUserToRoute: AuthenticatedUser => Route)(implicit authService: AuthService, executionContext: ExecutionContext): Route = {
+
+    onComplete(authService.authenticate(token)) {
+      case Success(userResult) => userResult match {
+        case ServiceResult.Success(team) =>
+          authUserToRoute(team)
+        case ServiceResult.Failure(Ex(ex)) =>
+          logger.error(s"OAuthService failed with exception $ex")
+          reject(CredentialsRejectedRejection(requestDescription))
+        case ServiceResult.Failure(_) =>
+          logger.error(s"OAuthService failed")
+          reject(CredentialsRejectedRejection(requestDescription))
       }
+
+      case Failure(ex) =>
+        logger.error(s"Error getting the authenticatedUser from the OAuthService for $requestDescription")
+        reject(InternalServerErrorRejection(requestDescription))
     }
+  }
+
+  def team(authenticatedUser: AuthenticatedUser, token: String, requestDescription: String, teamToRoute: (Team => Route))(implicit teamService: TeamService, executionContext: ExecutionContext): Route = {
 
-  def team(authenticatedUser: AuthenticatedUser, token: String, requestDescription: String)(implicit teamService: TeamService): Directive1[Team] = {
     authenticatedUser.username match {
       case Some(username) =>
-        teamService.getForUsername(username, token) match {
-          case ServiceResult.Success(team) => provide(team)
-          case ServiceResult.Failure(_)    => reject(TeamNotFoundRejection(requestDescription))
+        onComplete(teamService.getForUsername(username, token)) {
+          case Success(teamResult) => teamResult match {
+            case ServiceResult.Success(team) =>
+              teamToRoute(team)
+            case ServiceResult.Failure(NotFound) =>
+              logger.error("AuthenticatedUser does not have a team {}", authenticatedUser)
```

Comment:
Use only one formatting technique. Either string interpolation or placeholders. But do not mix them.